Populate README and remove duplicate (#30)
NCAS-CMS/cf-plot: README.md

# cf-plot

## Code-light plotting for earth science and aligned research

### Overview

**cf-plot allows you to produce and customise publication-quality contour,
vector, line and more plots with the power of Python,
[matplotlib](https://matplotlib.org/),
[Cartopy](https://scitools.org.uk/cartopy/docs/latest/) and
[cf-python](https://ncas-cms.github.io/cf-python/), in as few lines of code
as possible.**

It is designed to be a useful visualisation tool for environmental, earth

> and aligned science, for example to facilitate climate and meteorological

Reviewer: ```suggestion
and aligned sciences, for example to facilitate climate and meteorological
```

research. cf-plot is developed and maintained by the
[NCAS-CMS](https://cms.ncas.ac.uk/index.html) group, part of
[NCAS](https://ncas.ac.uk/).

### Brief Demonstration

> In as little as four lines of Python including imports, using `cf-plot` you can

Reviewer: ```suggestion
In as little as four lines of Python including, imports and file reading, using `cf-plot` you can
```

for example produce a contour plot showing a 2D subspace of a netCDF dataset:

```python
import cf
import cfplot as cfp
f = cf.read('<dataset name>.nc')[0]  # picks out a read-in field of the dataset
cfp.con(f.subspace(time=<chosen time value>))  # creates a contour plot of the field at that time value
```

### Examples Gallery

> A gallery of outputs made with cf-plot, showcasing a range of plotting
> possibilities with links to relevant documentation pages and to example code,
> can be found
> [on this dedicated page within the documentation](http://ajheaps.github.io/cf-plot/gallery.html),
> as illustrated in this (static) image:

Reviewer: ```suggestion
The [gallery of outputs made with cf-plot](http://ajheaps.github.io/cf-plot/gallery.html) showcases a range of plotting
possibilities with links to relevant documentation pages and to example code,
as illustrated in this (static) image:
```

Reviewer: ... thinking more, maybe you could move the image to the top of the README, so that it gets seen by the don't-scroll-down crowd. Just a thought.

Author: I did wonder whether this would be best, but also wanted to highlight to *that* crowd how few lines are required to get a plot out and put that first. However, now you mention it, they are more likely to scroll further past an image, so I agree with your suggestion. Thanks, I'll promote it to the first section, then mark this comment as resolved.

Author: Actually, re-reading I see you were probably instead proposing to move the image only, not that whole section. Might be best, good thinking. I'll do that and point to it from the 'Examples Gallery' section, left in the same place.

Author: See [3cb767e](https://github.com/NCAS-CMS/cf-plot/pull/30/commits/3cb767e65d6d69c86312b03d814091a1a5870e51) where I've moved the image (only) to the top.

![cf-plot example gallery of plots](docs/media/cf_gallery_image.png)

### Documentation

See [the cf-plot homepage](http://ajheaps.github.io/cf-plot)
(`http://ajheaps.github.io/cf-plot`) for the full online documentation.

### Installation

To install cf-plot with its required dependencies, you can use `pip`:

```bash
pip install cf-python
pip install cf-plot
```

or you can use `conda` (or similar package managers such
as `mamba`) as follows (or equivalent):

```bash
conda install -c ncas -c conda-forge cf-python cf-plot udunits2

> conda install -c conda-forge mpich esmpy

Author: Oh fair. I was just copying what Andy has on the ['Download/Install' page](Download/Install) without thinking over it much. I'll remove the line here and make sure it is clearer on that page that the second line of installs is optional.

> conda install -c conda-forge mpich esmpy

Author: See [3cb767e](https://github.com/NCAS-CMS/cf-plot/pull/30/commits/3cb767e65d6d69c86312b03d814091a1a5870e51).

```

More detail about installation is provided on the
[installation page](http://ajheaps.github.io/cf-plot/download.html)
(`http://ajheaps.github.io/cf-plot/download.html`)
of the documentation.

### Contributing

Everyone is welcome to contribute to cf-plot, however the NCAS-CMS team reserve
the right to only accept changes which they consider to improve the codebase

> and align with future ambitions for the library.

Author: Indeed, I just thought it might be wise to indicate that we can't be bound to accepting any old changes that someone might propose. Notably, in a previous project I worked on, someone did months of work towards a PR without checking in (formally/regularly) that it was what was wanted by the core project team, and after difficult conversations the PR was closed and not incorporated and I imagine someone felt very disgruntled. I think mainly I wanted to indicate that any work contributed should be aligned with an Issue raised and signed off by myself or someone else relevant in NCAS-CMS, but probably if we say something along those lines it should be in the more detailed Contributing guidance I will soon write up (or specifically, copy from another good OS project).

In short, I agree with your proposal but think we should add something along the above lines to the more detailed guidance, to prevent anyone possibly doing some work towards contributions which we can't accept.

Committing your suggestion (there's a typo but I'll fix that up in a final commit). Thanks.

Contributing guidelines will be added to the repository shortly.

### Help: Issues, Questions, Feature Requests, etc.

For any queries, see the
[guidance page](http://ajheaps.github.io/cf-plot/issues.html)

> (`http://ajheaps.github.io/cf-plot/issues.html`).

Reviewer: Surely https://github.com/NCAS-CMS/cf-plot/issues ?

Author: I haven't yet moved the documentation source and build across to the NCAS-CMS space, so technically that is the canonical link at present, though Andy has set up redirection for now. I was planning to update the links once I had done the move, but you're right it might be best just put the eventual ones in for now (will be less confusing to refer to the same user/org space as the repo belongs to). Thanks.

Author: Oh, realising I misinterpreted what you said in my reply above (ignore it).

Andy's current guidance page states people should email me directly. I have started to initiate the Issue Tracker for this instead, but haven't had a chance to update the guidance. Until I update that I will leave this pointer so as not to make contradictory statements RE issue reporting.

Author: Hope that makes sense and is OK to you. I'll leave this unresolved in case you want to say otherwise :slightly_smiling_face:
